File: nesne/banka.py
class Account:
    accounts = []

    def __init__(self, account_number, owner, balance):
        self.account_number = account_number
        self.owner = owner
        self.balance = balance
        Account.accounts.append(self)
        Bank.track_transaction(f"Hesap olusturuldu: {account_number}")

    def deposit(self, amount):
        if amount > 0:
            self.balance += amount
            print(f"{amount} TL yatirildi. Yeni bakiye: {self.balance} TL")
            Bank.track_transaction(f"{self.account_number} hesabina {amount} TL yatirildi.")
        else:
            print("Gecersiz yatirma tutari.")

    def withdraw(self, amount):
        if amount > 0 and amount <= self.balance:
            self.balance -= amount
            print(f"{amount} TL cekildi. Yeni bakiye: {self.balance} TL")
            Bank.track_transaction(f"{self.account_number} hesabindan {amount} TL cekildi.")
        else:
            print("Yetersiz bakiye veya gecersiz tutar.")

    def view_balance(self):
        print(f"Hesap Sahibi: {self.owner}")
        print(f"Hesap Numarasi: {self.account_number}")
        print(f"Bakiye: {self.balance} TL")

class Bank:
    transaction_history = []

    @staticmethod
    def display_bank_info():
        print(f"Bankamizda toplam {len(Account.accounts)} hesap bulunmaktadir.")
        print("Toplam islemler:")
        for transaction in Bank.transaction_history:
            print(transaction)

    @staticmethod
    def track_transaction(description):
        Bank.transaction_history.append(description)

if __name__ == "__main__":
    while True:
        print("\n--- Banka Hesap Yonetim Sistemi ---")
        print("1. Hesap Ac")
        print("2. Para Yatir")
        print("3. Para Cek")
        print("4. Bakiye Goruntule")
        print("5. Tum Hesaplari Goruntule")
        print("6. Cikis")
        
        secim = input("Lutfen bir secim yapiniz: ")
        
        if secim == "1":
            account_number = input("Hesap Numarasi: ")
            owner = input("Hesap Sahibi: ")
            balance = float(input("Baslangic Bakiyesi: "))
            Account(account_number, owner, balance)
            print("Hesap basariyla olusturuldu.")
        
        elif secim == "2":
            account_number = input("Hesap Numarasi: ")
            amount = float(input("Yatirilacak Tutar: "))
            for account in Account.accounts:
                if account.account_number == account_number:
                    account.deposit(amount)
                    break
            else:
                print("Hesap bulunamadi.")
        
        elif secim == "3":
            account_number = input("Hesap Numarasi: ")
            amount = float(input("Cekilecek Tutar: "))
            for account in Account.accounts:
                if account.account_number == account_number:
                    account.withdraw(amount)
                    break
            else:
                print("Hesap bulunamadi.")
        
        elif secim == "4":
            account_number = input("Hesap Numarasi: ")
            for account in Account.accounts:
                if account.account_number == account_number:
                    account.view_balance()
                    break
            else:
                print("Hesap bulunamadi.")
        
        elif secim == "5":
            Bank.display_bank_info()
        
        elif secim == "6":
            print("Cikis yapiliyor...")
            break
        
        else:
            print("Gecersiz secim, tekrar deneyin.")

File: nesne/banka.java
import java.util.*;

// Account Sınıfı
class Account {
    private String accountNumber;
    private String owner;
    private double balance;
    
    public static List<Account> accounts = new ArrayList<>();

    public Account(String accountNumber, String owner, double balance) {
        this.accountNumber = accountNumber;
        this.owner = owner;
        this.balance = balance;
        accounts.add(this);
    }

    public void deposit(double amount) {
        if (amount > 0) {
            balance += amount;
            System.out.println(amount + " TL hesaba yatirildi. Yeni bakiye: " + balance + " TL");
            Bank.trackTransaction(accountNumber + " nolu hesaba " + amount + " TL yatirildi.");
        } else {
            System.out.println("Gecersiz yatirma miktari.");
        }
    }

    public void withdraw(double amount) {
        if (amount > 0 && balance >= amount) {
            balance -= amount;
            System.out.println(amount + " TL cekildi. Yeni bakiye: " + balance + " TL");
            Bank.trackTransaction(accountNumber + " nolu hesaptan " + amount + " TL cekildi.");
        } else {
            System.out.println("Yetersiz bakiye veya gecersiz cekim miktari.");
        }
    }

    public void viewBalance() {
        System.out.println("Hesap Sahibi: " + owner);
        System.out.println("Hesap Numarasi: " + accountNumber);
        System.out.println("Bakiye: " + balance + " TL");
    }
}

// Bank Sınıfı
class Bank {
    public static List<String> transactionHistory = new ArrayList<>();

    public static void displayBankInfo() {
        System.out.println("Bankamiza hosgeldiniz. Tum hesaplar guven altindadir.");
        System.out.println("Toplam Hesap Sayisi: " + Account.accounts.size());
    }

    public static void trackTransaction(String description) {
        transactionHistory.add(description);
    }
}

// Test Kodları
public class BankSystem {
    public static void main(String[] args) {
        Account acc1 = new Account("12345", "Ali Yilmaz", 1000);
        Account acc2 = new Account("67890", "Ayse Demir", 2000);

        acc1.viewBalance();
        acc1.deposit(500);
        acc1.withdraw(300);
        acc1.viewBalance();
        
        acc2.viewBalance();
        acc2.withdraw(2500);

        Bank.displayBankInfo();
        
        System.out.println("\nIslem Gecmisi:");
        for (String transaction : Bank.transactionHistory) {
            System.out.println(transaction);
        }
    }
}
